### PR TITLE
Mg/pca and profilers

### PR DESCRIPTION
**Pull request by michael-gemmell**
1. PCA for more efficient optimisation of profiles
2. Different version of PlasmaProfiler has been tweaked to work with BDA workflow
3. Profilers have been standardised / streamlined
4. Settings for the above components are in configs where possible

### CODE REVIEW

File: indica/workflows/pca.py
Comment (marcosertoli) [inline indica/workflows/pca.py:1]: A lot of lines of code but no docstrings.

Could you add some details to each function/method?

If you move the profilers to indica/profilers, would it make sense to move this there as well?

Also, the /workflows folder is getting too crowded... Can we reorganise this somehow?

File: indica/profilers.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+from indica.utilities import format_coord
+from indica.utilities import format_dataarray
+
+
+def gaussian(x, A, B, x_0, w):
+    return (A - B) * np.exp(-((x - x_0) ** 2) / (2 * w**2)) + B
+
+
+class Profiler(ABC):
```
[the remark this answers is not in the document]
Comment (michael-gemmell) [reply indica/profilers.py:18]: I would suggest: 
profiler_abstract
profiler_gauss
profiler_other

File: indica/configs/workflows/plasma_profiler/default.yaml
[the remark this answers is not in the document]
Comment (michael-gemmell) [reply indica/configs/workflows/plasma_profiler/default.yaml:1]: Yeah I guess whicher workflow wants to use this class can setup the profilers as they like.

File: indica/workflows/pca.py
Comment (michael-gemmell) [reply indica/workflows/pca.py:1]: Maybe I should quarantine my BDA specific code in it's own directory

File: indica/workflows/plasma_profiler.py
[the remark this answers is not in the document]
Comment (michael-gemmell) [reply indica/workflows/plasma_profiler.py:1]: It can but plasma.py is already massive was my thinking.

**Pull request closed without merging**